### PR TITLE
Exposing the Tempo API using the gateway

### PR DESCRIPTION
**Pull request by rubenvp8510**
This depends on https://github.com/observatorium/api/pull/575

### CODE REVIEW

File: internal/manifests/gateway/gateway.go [hunk 1]
```diff
@@ -168,6 +168,8 @@ func deployment(params manifestutils.Params, rbacCfgHash string, tenantsCfgHash
 								fmt.Sprintf("--traces.write.endpoint=%s:4317", naming.ServiceFqdn(tempo.Namespace, tempo.Name, manifestutils.DistributorComponentName)),
 								fmt.Sprintf("--traces.read.endpoint=%s://%s:16686", httpScheme(params.Gates.HTTPEncryption),
 									naming.ServiceFqdn(tempo.Namespace, tempo.Name, manifestutils.QueryFrontendComponentName)),
+								fmt.Sprintf("--traces.tempo.endpoint=%s://%s:3200", httpScheme(params.Gates.HTTPEncryption),
```
Comment (andreasgerstmayr) [inline internal/manifests/gateway/gateway.go:171]: Let's use `manifestutils.PortHTTPServer` for the Tempo port, and also the relevant port constant from `manifestutils/constants.go` for the 2 other hardcoded ports above.

**Comment by pavolloffay** [comment]
Don't forget to bump the gateway version :)

**Comment by rubenvp8510** [comment]
@andreasgerstmayr Do you think we need to have a flat feature or a field in the CRD to enable or disable this feature?

**Comment by andreasgerstmayr** [comment]
> @andreasgerstmayr Do you think we need to have a flag feature or a field in the CRD to enable or disable this feature?

I think we don't need a new flag, because we already expose the same data (all traces) with a different API (tempo-query).

**Review by andreasgerstmayr — COMMENTED** [review]
Did you run the openshift e2e tests? Afaics the multitenancy test should fail, as it has an assert on the container arguments.
Can you also update gateway_test.go to add an assert for the new traces.tempo.endpoint CLI argument?

LGTM otherwise :+1:

**Comment by rubenvp8510** [comment]
> Did you run the openshift e2e tests? Afaics the multitenancy test should fail, as it has an assert on the container arguments. Can you also update gateway_test.go to add an assert for the new traces.tempo.endpoint CLI argument?
> 
> LGTM otherwise 👍

Oh yes you're right, I'll update openshift tests! Thanks for the review

**Comment by rubenvp8510** [comment]
@andreasgerstmayr fixed openshift test, also added verify traces for the traceql endpoint. and the assert of the argument

**Review by andreasgerstmayr — APPROVED** [review]
awesome, thanks! I'll merge after CI passes.

**Comment by rubenvp8510** [comment]
Thank youu!

**Pull request merged**